core/src/main/scala/flatgraph/Misc.scala

```diff
@@ -0,0 +1,26 @@
+package flatgraph
+
+import java.util.concurrent
+
+object Misc {
+
+  @volatile var _overrideExecutor: concurrent.ExecutorService => concurrent.ExecutorService = defaultExecutorProvider
+
+  def force_singlethreaded(): Unit = {
+    // this one is magic -- it can get garbage collected, no manual shutdown required!
+    this._overrideExecutor = (something: concurrent.ExecutorService) => concurrent.Executors.newSingleThreadExecutor()
+  }
+
+  def defaultExecutorProvider(current: concurrent.ExecutorService): concurrent.ExecutorService = {
+    current match {
+      case null =>
+        java.lang.Thread.currentThread() match {
+          case fjt: concurrent.ForkJoinWorkerThread => fjt.getPool
```

Comment:
why not import the classes rather than the `concurrent` namespace?
that would make this more readably IMO, especially given the mix of 'current and 'concurrent'